specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json

```diff
@@ -710,6 +710,253 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/basicPublishingCredentialsPolicies": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Returns whether Scm basic auth is allowed and whether Ftp is allowed for a given site.",
+        "description": "Description for Returns whether Scm basic auth is allowed and whether Ftp is allowed for a given site.",
+        "operationId": "WebApps_GetBasicPublishingCredentialsPolicies",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the app.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CsmPublishingCredentialsPoliciesCollection"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Publishing Credentials Policies": {
+            "$ref": "./examples/ListPublishingCredentialsPolicies.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/basicPublishingCredentialsPolicies/ftp": {
```

Comment:
For both PUT and GET patterns, this could be collapsed into one API path. The request and response schema for both SCM and FTP is the same. So you can model the URI as
`/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/basicPublishingCredentialsPolicies/{credentialsTypeId}`  where `credentialsTypeId` is a string enum parameter (in the path). 

It would reduce redundancy in the sepc.

Reply:
Our swagger definitions are auto-generated based on routes we define in our source code. Unfortunately I don't think we are currently able to model enum constrain in our routes and in order to keep this definition consistent with what is auto-generated I think it's best to leave it as is. I have however created a tracking item to investigate modeling enum constraints in our definitions so when we are able I will update the definition to reduce redundancy.